tests/subsAndMods.test.ts

```diff
@@ -0,0 +1,361 @@
+  if (queue.testAccess === undefined) {
+    throw new Error("testAccess is undefined");
+  }
+
+  const getList: QueueDataMap<OnlineOfflineList> = await queue.list();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(0);
+    expect(list.offline).toHaveLength(0);
+  });
+
+  const chatter1 = buildChatter("chatter1", "chatter1", false, false, false);
+  const chatter2 = buildChatter("chatter2", "chatter2", false, false, false);
+  const chatter3 = buildChatter(
+    "chatter3",
+    "chatter3",
+    false,
+    false,
+    false,
+    "chatter3"
+  );
+
+  const level1 = "D36-010-5YF";
+  const level2 = "MY2-H2M-DSG";
+  const level3 = "GBJ-6QY-P8G";
+
+  expect(queue.add(level1, chatter1)).toContain("has been added to the queue");
+  expect(queue.add(level2, chatter2)).toContain("has been added to the queue");
+  expect(queue.add(level3, chatter3)).toContain("has been added to the queue");
+
+  return {
+    queue,
+    chatter1,
+    chatter2,
+    chatter3,
+    twitch,
+  };
+}
+
+test("fetch subs", async () => {
+  const { queue, chatter1, chatter2, chatter3 } = await setupQueue();
+  if (queue.testAccess === undefined) {
+    throw new Error("testAccess is undefined");
+  }
+
+  // Add a subscriber to the "backend"
+  simSetSubscribers([chatter1]);
+
+  // Add our mock users to chat
+  simSetChatters([chatter1, chatter2, chatter3]);
+
+  // Ensure there are no subs yet
+  let getList: QueueDataMap<OnlineOfflineList> = await queue.sublist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(0);
+  });
+
+  // Wait 15 minutes, check for chatter1 as a sub
+  await simAdvanceTime(15 * 60 * 1000);
+
+  // Ensure our sub is now online
+  getList = await queue.sublist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(1);
+    expect(list.online.at(0)?.submitter).toEqual({
+      ...chatter1,
+      isSubscriber: true,
+    });
+  });
+});
+
+test("fetch mods", async () => {
+  const { queue, chatter1, chatter2, chatter3 } = await setupQueue();
+  if (queue.testAccess === undefined) {
+    throw new Error("testAccess is undefined");
+  }
+
+  // Add a moderator to the "backend"
+  simSetModerators([chatter2]);
+
+  // Add our mock users to chat
+  simSetChatters([chatter1, chatter2, chatter3]);
+
+  // Ensure there are no mods yet
+  let getList: QueueDataMap<OnlineOfflineList> = await queue.modlist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(0);
+  });
+
+  // Wait 15 minutes, check for chatter2 as a mod
+  await simAdvanceTime(15 * 60 * 1000);
+
+  // Ensure our mod is now online
+  getList = await queue.modlist();
+  queue.testAccess((data) => {
+    const list = getList(data);
+    expect(list.online).toHaveLength(1);
+    expect(list.online.at(0)?.submitter).toEqual({ ...chatter2, isMod: true });
```

Comment:
Checking `isMod` here has the same issues as checking `isSubscriber`.

Reply:
Switched to ID here too.